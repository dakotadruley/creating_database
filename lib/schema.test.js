const { Schema } = require('./Schema.js');

//  updated
describe('Schema', () => {
  it('check the object to see if a correct schema', () => {
    const dog = {
      name: 'spot',
      age: 5,
      weight: '20 lbs'
    };

    const schema = new Schema({
      name: {
        type: String,
        required: true
      },
      age: {
        type: Number,
        required: true
      },
      weight: {
        type: String
      }
    });

    expect(schema.validate(dog)).toEqual({
      name: 'spot',
      age: 5,
      weight: '20 lbs'
    });
  });
  
});

it('throws an error when validating an object does not fallow the given schema', () => {
  const dog = {
    name: 456,
    age: 'bad',
    weight: 15,
  };

  const schema = new Schema({
    name: {
      type: String,
      required: true
    },
    age: {
      type: Number,
      required: true
    },
    weight: {
      type: String
    }
  });

  expect(() => schema.validate(dog)).toThrowErrorMatchingSnapshot();
});


  

