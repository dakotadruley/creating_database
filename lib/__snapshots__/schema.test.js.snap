// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws an error when validating an object does not fallow the given schema 1`] = `"Invalid schema >> Error: Cannot cast >>bad<< to Number"`;
